src/app/stats/stats.component.ts

```diff
@@ -7,9 +34,78 @@ import { Component, OnInit } from '@angular/core';
 })
 export class StatsComponent implements OnInit {
 
-  constructor() { }
+  inProgress = false;
+  nodes: Array<GraphVertex> = [];
+  links: Array<GraphEdge> = [];
 
-  ngOnInit() {
+  constructor(
+    private readonly blockService: BlockTreeService,
+    private readonly progress: ProgressService,
+  ) {
+    this.progress.progress.subscribe(v => this.inProgress = v);
   }
 
+  // slotPoll() {
+  //   return interval(SLOT_DURATION_MILLISECONDS)
+  //     .pipe(
+  //       switchMap(() => {
+  //         this.progress.startProgress();
+  //         return this.blockService.getBlockTree();
+  //       }), 
+  //       // distinctUntilChanged(), // only emit and redraw graph if the value has actually changed
+  //     );
+  // }
+
+  ngOnInit() {
+    this.blockService.getBlockTree().subscribe(async (res: BlockTreeResponse) => {
+      const tree = this.blockService.sortTree(res.getTreeList());
+      const existingParentBlocks = {};
+      this.nodes = tree.map((node, idx) => {
+        const blockRoot = this.blockService.toHexString(node.getBlockRoot());
+        existingParentBlocks[blockRoot] = true;
+
+        let participatedVotesString = Math.floor(node.getParticipatedVotes() / MAX_DEPOSIT_AMOUNT).toString();
+        if (participatedVotesString.indexOf('.') === -1) {
+          participatedVotesString += '.';
+        }
+        while (participatedVotesString.length < participatedVotesString.indexOf('.') + 15) {
+          participatedVotesString += '0';
+        }
+        let totalVotesString = Math.floor(node.getParticipatedVotes() / MAX_DEPOSIT_AMOUNT).toString();
```

Comment:
```suggestion
        let totalVotesString = Math.floor(node.getTotalVotes() / MAX_DEPOSIT_AMOUNT).toString();
```

I'm assuming. Didn't follow the logic thoroughly, but the logic for `participatedVotesString` and `totalVotesString` is currently identical